examples/wip/attribute-transition/app.js

```diff
@@ -0,0 +1,131 @@
+/* global document, window, console */
+/* eslint-disable no-console */
+import React, {Component} from 'react';
+import {render} from 'react-dom';
+import MapGL from 'react-map-gl';
+import DeckGL, {ScatterplotLayer} from 'deck.gl';
+import * as d3 from 'd3-ease';
+
+import PointGenerator from './point-generator';
+
+// Set your mapbox token here
+const MAPBOX_TOKEN = process.env.MapboxAccessToken; // eslint-disable-line
+
+class Root extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this._pointGenerator = new PointGenerator({
+      center: [-122.44, 37.75],
+      distance: [0.064, 0.05],
+      radiusRange: [5, 100],
+      countRange: [1000, 10000]
+    });
+
+    this.state = {
+      viewport: {
+        latitude: 37.75,
+        longitude: -122.44,
+        zoom: 12,
+        width: window.innerWidth,
+        height: window.innerHeight
+      },
+      data: this._pointGenerator.points,
+      pointsUpdated: 0,
+      radiusUpdated: 0,
+      colorsUpdated: 0
+    };
+
+    this._onViewportChange = this._onViewportChange.bind(this);
+    this._resize = this._resize.bind(this);
+    this._randomizeCount = this._randomizeCount.bind(this);
+    this._randomizePositions = this._randomizePositions.bind(this);
+    this._randomizeRadius = this._randomizeRadius.bind(this);
+    this._randomizeColors = this._randomizeColors.bind(this);
+  }
+
+  componentDidMount() {
+    window.addEventListener('resize', this._resize);
+  }
+
+  _resize() {
+    this._onViewportChange({
+      width: window.innerWidth,
+      height: window.innerHeight
+    });
+  }
+
+  _onViewportChange(viewport) {
+    this.setState({
+      viewport: {...this.state.viewport, ...viewport}
+    });
+  }
+
+  _randomizeCount() {
+    this._pointGenerator.randomizeCount();
+    this.setState({data: this._pointGenerator.points});
+  }
+
+  _randomizePositions() {
+    this._pointGenerator.randomizePositions();
+    this.setState({positionsUpdated: Date.now()});
+  }
+
+  _randomizeRadius() {
+    this._pointGenerator.randomizeRadius();
+    this.setState({radiusUpdated: Date.now()});
+  }
+
+  _randomizeColors() {
+    this._pointGenerator.randomizeColors();
+    this.setState({colorsUpdated: Date.now()});
+  }
+
+  render() {
+    const {viewport, positionsUpdated, radiusUpdated, colorsUpdated} = this.state;
+
+    const layer = new ScatterplotLayer({
+      data: this._pointGenerator.points,
+      getPosition: d => d.position,
+      getColor: d => d.color,
+      getRadius: d => d.radius,
+      updateTriggers: {
+        getPosition: positionsUpdated,
+        getRadius: radiusUpdated,
+        getColor: colorsUpdated
+      },
+      transition: {
```

Comment:
Should we call it `transitions` (ref. `updateTriggers`)

An interesting question is how should this "deep object" i.e. `props.transitions` be diffed?  My suggestion is that we specify that we ignore it completely during diffing (for max perf) and only consult it when needed. That means that changing a value here will not trigger any change to an ongoing transition.

Also we will likely want to add prop transitions to this object as well, so that we have a unified approach.